### DISCUSSION TITLE
API Documentation

### DISCUSSION THREAD

**Discussion by p5** [withastro/starlight#1715] [Feature Requests]
### What version of `starlight` are you using?

0.0.0

### What is your idea?

[Redoc](https://github.com/redocly/redoc) is a library / tool for turning Swagger and OpenAPI specs into nice UIs.
It comes as a HTML tag or React component.

Since Starlight is used for documentation, it would also be good to add native support for rendering API documentation given the JSON spec file.

### Why is this feature necessary?

We have text documentation, but no way of natively (I.E. without introducing additional external dependencies or defining in Markdown) displaying API documentation.  

### Participation

- [ ] I am willing to submit a pull request for this feature.

**Comment by HiDeoo** [comment]
Thanks for your request :raised_hands:

Only my personal opinion, but I think having Swagger and OpenAPI support natively in Starlight could feel a bit too opinionated and would add a lot of maintenance overhead to the project. I think a community [plugin](https://starlight.astro.build/resources/plugins/) would be a better approach for this specific use case.

As far as I know, there is not yet any plugin for Redoc but there is a [`starlight-openapi`](https://starlight-openapi.vercel.app/) one.

**Comment by HiDeoo** [comment]
We are particulary happy with the current approach relying on [Starlight plugins](https://starlight.astro.build/resources/plugins/) to handle such use cases, e.g. [`starlight-openapi`](https://starlight-openapi.vercel.app/), [`starlight-typedoc`](https://starlight-typedoc.vercel.app/), etc.

It avoids adding a lot of maintenance burden to the core Starlight codebase, while allowing users to extend Starlight with their own custom features or 3rd-party plugins, and plugin can move at their own pace.

Considering this, I'm going to close this discussion as Redoc support is not something we plan to add to Starlight core, but rather something that could be implemented as a plugin.
